vars/buildDotNetCore.groovy

```diff
@@ -33,8 +33,12 @@ def call(Map config=[:]) {
         config['buildClosure']()
       }
 
-      stage('Build test image') {
-        build.buildTestImage(DOCKER_REGISTRY_CREDENTIALS_ID, DOCKER_REGISTRY, repoName, BUILD_NUMBER, tag)
+      if (fileExists('./docker-compose.test.yaml')) {
+        stage('Build test image') {
+          build.buildTestImage(DOCKER_REGISTRY_CREDENTIALS_ID, DOCKER_REGISTRY, repoName, BUILD_NUMBER, tag)
+        }
+      } else {
+      echo("docker-compose.test.yaml not found, skipping test run")
```

Comment:
I wouldn't bother with the else statement now.  Because it's moved to the top level global variable, it won't be written to a stage anyway so isn't going to be as useful as when it was in the class.